common/common-rest/src/main/java/io/servicecomb/common/rest/codec/param/FormProcessorCreator.java

```diff
@@ -69,6 +71,44 @@ public FormProcessorCreator() {
   @Override
   public ParamValueProcessor create(Parameter parameter, Type genericParamType) {
     JavaType targetType = TypeFactory.defaultInstance().constructType(genericParamType);
+
+    if (isPart(parameter)) {
+      return new PartProcessor(parameter.getName(), targetType);
+    }
     return new FormProcessor(parameter.getName(), targetType);
   }
+
+  private boolean isPart(Parameter parameter) {
+    return new FileProperty().getType().equals(((FormParameter) parameter).getType());
+  }
+
+  private static class PartProcessor extends AbstractParamProcessor {
+    PartProcessor(String paramPath, JavaType targetType) {
+      super(paramPath, targetType);
+    }
+
+    @Override
+    public Object getValue(HttpServletRequest request) throws Exception {
+      @SuppressWarnings("unchecked")
+      Map<String, Object> forms = (Map<String, Object>) request.getAttribute(RestConst.FORM_PARAMETERS);
+      if (forms != null) {
+        return convertValue(forms.get(paramPath), targetType);
+      }
+
+      return request.getPart(paramPath);
+    }
+
+    @Override
+    public void setValue(RestClientRequest clientRequest, Object arg) throws Exception {
+      String[] filenames = (String[]) arg;
+      for (String filename : filenames) {
+        clientRequest.attach(paramPath, filename);
```

Comment:
attach multiple times
but implementation is Map<String, String>, not Map<String, List<String>>

Reply:
it's actually a MultiValueMap, which is a Map<String, List>

Reply:
RestClientRequestImpl .uploads :

Map<String, String> uploads = new HashMap<>();